wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Use Updated and Null Proof `CommentModel` Class (`breaking`)

Parent [FluxC#2799](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2799)
Accompanying [FluxC#2870](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870)

This PR accompanying the above FluxC PR, and uses an updated and null-proof [CommentModel.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java#L4) class.

FYI: This change is `breaking`, meaning that any client that depended on the `Comment` model should update to the new APIs. As such, this change is inherently `risky`, meaning that there are compile-time changes associated with this change and thus needs testing to verify correctness, both on the library's and client's side.

PS: The [CommentModel.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java#L4) model class is shared between the deprecated `CommentStore` and the `CommentsStore`, thus testing any `CommentsStore` related functionality, that is, in addition to testing any the  `CommentStore` related functionality (shouldn't be many) is highly recommended.

-----

@wzieba let me know how I did on this PR, that is, in terms of updating the store/sqlutils classes and adding nullability annotations on them and their main [CommentModel.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java#L4) model class, in order to utilize the nullability annotations added on its fields and thus make its use null proof. If you also feel that this is a good way to proceed with such PRs, I'll then use this pattern across the board and follow a similar approach on every such PR.

-----

FluxC Update List:

1. [Update fluxc version to pr hash (2870)](https://github.com/wordpress-mobile/WordPress-Android/pull/19389/commits/ab1afaa82ef543195f05599fdf532e4427f9188b)
2. [Update fluxc version to pr hash (2870)](https://github.com/wordpress-mobile/WordPress-Android/pull/19389/commits/e1a6a978b2f4ffcaea893ab9cfddd5343623d0a4)
3. [Update fluxc version to trunk hash (2870)](https://github.com/wordpress-mobile/WordPress-Android/pull/19389/commits/1a3ee3db3715bb281014df1ea52a6bf4105fd9d0)

Test List:

1. [Update get likes use case test to updated on cmt likes changed](https://github.com/wordpress-mobile/WordPress-Android/pull/19389/commits/98f9f11115754c4607a62a9a2c15d5930e7a1b6f)

-----

## To Test (`REST`):

- Smoke test any `CommentStore` and `CommentsStore` related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

## To Test (`XMLRPC`):

- Create a new self-hosted WordPress site for XMLRPC testing purposes ([jurassic.ninja](https://jurassic.ninja/)).
- Smoke test any `CommentStore` and `CommentsStore` related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

<details>
    <summary>1. Comment Details Screen from Comments Screen [CommentDetailFragment.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Comments` screen and tap on any comment.
- Verify that the `Comment Details` screen is shown and functioning as expected.
- For example, try replying, approving, marking as spam, liking, trashing, editing or copying a comment's link address. Additionally, make sure to verify that you can still swipe left-right to navigate between comments.

</details>

<details>
    <summary>2. Comment Details Screen from Notification Tab [CommentDetailFragment.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Notifications` tab, then its `COMMENTS` sub-tab, and tap on any comment.
- Verify that the `Comment Details` screen is shown and functioning as expected.
- For example, try replying to a comment. Additionally, make sure to verify that you can still swipe left-right to navigate between comments.

</details>

<details>
    <summary>3. Edit Comment Screen from Comments Screen [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Comments` screen and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its `Comment` itself.

</details>

<details>
    <summary>4. Edit Comment Screen from Notification Tab [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Notifications` tab, then its `COMMENTS` sub-tab, and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its `Comment` itself.

</details>

<details>
    <summary>5. Reader Comments Screen [ReaderCommentListActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Reader` tab, then its `DISCOVER` sub-tab, and tap on any post.
- Tap on the `Comments` button at the bottom of the post.
- Verify that the `Reader Comments` screen is shown and functioning as expected.
- For example, try (un)following, enabling/disabling in-app notifications, replying to post, and then sharing, replying or liking a specific comment.

</details>

-----

# Merge instructions:

1. [x] Wait for the accompanying [FluxC#2870](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870) PR to be merged.
4. [x] Update `wordPressFluxCVersion` to point to the `trunk` hash that includes the above solution.
5. [x] Remove the `[Status] Not Ready for Merge` label.
6. [x] Merge this PR.

PS: This PR will be failing until [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2868) base PR is merged to `trunk` and the accompanying [FluxC#2870](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870) gets updated with the latest `trunk` version of FluxC.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
